jina/executors/indexers/keyvalue.py

```diff
@@ -30,13 +30,17 @@ def __init__(self, path, mode):
 
         def close(self):
             """Close the file."""
-            self.body.close()
-            self.header.close()
+            if not self.body.closed:
+                self.body.close()
+            if not self.header.closed:
+                self.header.close()
```

Comment:
Looks like a smell. Why is `close` called multiple times?

Reply:
We have body and header I think? Both need to be closed. What would be the best way?

Reply:
It might be called several times since it depends on what state the Indexer is in when it's closed and whether the `delete_on_dump` is set to `True`. One of the tests to which I added `del_on_dump: true` in this PR exhibited this error